### PR TITLE
Treat pytest collection errors as terminal

### LINKED ISSUE #485: Retry constructs invalid pytest node ID for collection errors (empty scope)
## Summary

When bktec retries a pytest collection error (e.g. an import failure), it constructs an invalid node ID because the scope is empty. This causes `no tests ran` on the retry attempt.

The test still correctly fails overall — the collection error is reported as a failure and bktec exits non-zero. But the retry is wasted and the error message is confusing.

## Root cause

In [`internal/runner/pytest.go` line 106](https://github.com/buildkite/test-engine-client/blob/v2.4.0/internal/runner/pytest.go#L106), bktec constructs retry node IDs by joining scope and name:

```go
Path: fmt.Sprintf("%s::%s", test.Scope, test.Name)
```

For normal test failures this produces valid node IDs like `tests/foo.py::test_bar`.

For collection errors, the scope is empty — and correctly so. A collection error's node ID is just the file path (e.g. `tests/foo.py`) with no `::` separator. When the test-collector-python plugin splits on `::`, it gets `scope=""` and `name="tests/foo.py"`. The unconditional join produces:

```
::tests/foo.py
```

This is not a valid pytest node ID, so pytest reports `no tests ran`.

## Suggested fix

Handle empty scope when constructing the retry path:

```go
if test.Scope != "" {
    path = fmt.Sprintf("%s::%s", test.Scope, test.Name)
} else {
    path = test.Name
}
```

This would produce `tests/foo.py` for collection errors, which is a valid pytest node ID. Pytest would attempt to collect the file again — it would still fail (the import error can't be fixed by re-running), but the error message would be clear and the retry behavior would be consistent.

## Context

This issue was discovered while investigating [buildkite/test-collector-python#106](https://github.com/buildkite/test-collector-python/issues/106), where collection errors were silently dropped from the JSON report. The fix in [buildkite/test-collector-python#108](https://github.com/buildkite/test-collector-python/pull/108) adds collection errors to the report, which surfaces this pre-existing issue in bktec's retry logic.

## Reproduction

https://github.com/altana-ai/bktec-collection-error-repro — the third scenario (`03-bktec-retry`) demonstrates the invalid node ID on retry after applying the test-collector-python fix.

### PR DESCRIPTION
**Pull request by pda**
### Description

When `buildkite-test-collector` reports a pytest collection error, the JSON result has an empty `scope` and a file-only `name` such as `tests/test_broken.py`. bktec was rebuilding every pytest JSON result path as `scope::name`, producing invalid retry selectors like `::tests/test_broken.py`.

This fixes the selector and treats explicitly tagged pytest collection errors as terminal errors outside of tests. Retrying individual pytest node IDs cannot fix import/syntax/collection failures, so this aligns pytest with the Go build-failure behavior from #540.

### Context

`buildkite/test-collector-python#108` made collection errors visible in the JSON report and tags them with `test.pytest_collection_error=true`. bktec can now use that explicit tag instead of guessing from empty scope alone.

### Changes

- Preserve file-only pytest node IDs when Test Engine JSON results have an empty scope.
- Mark tagged pytest collection errors as `RunStatusError`, which uses the existing terminal retry guard.
- Reuse the same JSON result mapping for `pytest-pants`.
- Add parser regression tests for tagged collection errors, untagged empty-scope results, and normal scoped pytest node IDs.
- Add a real pytest fixture with a collection-time import error so CI exercises actual pytest/collector output.

### Testing

- `go test ./internal/runner -run '^TestPytestRun_CollectionError$' -v` in a temporary `mise`/`uv` virtualenv with `pytest` and `buildkite-test-collector>=1.3.0` installed
- `go test ./internal/runner -run 'TestPytestRunParseJSON|TestPytestPathFromTestEngineResult'`
- `go test ./internal/runner -run '^TestPytestPantsRun_TestFailed$'`
- `go test ./internal/command -run 'TestRunTestsWithRetry_RunResultError|TestRunTestsWithRetry_GoTestBuildFailed'`

I also attempted `go test ./internal/runner ./internal/command`, but this local environment is missing some runner dependencies such as `cypress` and the broader runner integration suite is not green here.

### Related

- Fixes #485
- Related to buildkite/test-collector-python#108
- Aligns with #540

### CODE REVIEW

File: internal/runner/pytest.go [hunk 1]
```diff
@@ -143,20 +143,41 @@ func (p Pytest) runParseJSON(result *RunResult) error {
...
+		Format:     plan.TestCaseFormatExample,
+		Scope:      test.Scope,
+		Name:       test.Name,
+		// pytest can execute individual tests using node IDs, which are filenames,
+		// class names (if any), and functions separated by `::`.
+		// Ref: https://docs.pytest.org/en/6.2.x/usage.html#nodeids
+		Path: path,
+	}, test.Result)
+
+	if test.Tags[pytestCollectionErrorTag] == "true" {
```
[the remark this answers is not in the document]
Comment (pda) [reply internal/runner/pytest.go:177]: 🤖 Good catch — pytest collection errors can exit 2 before the JSON parsing path ran. Fixed in 5752935 by allowing the JSON collector path to parse exit 2 results while preserving the original command error, and added a regression test for tagged collection-error JSON on exit 2.

Verified with:

```sh
go test ./internal/runner -run 'TestPytestRun_JSONExit2ParsesCollectionError|TestPytestRunParseJSON_CollectionError|TestPytestRun_CollectionError'
```

I also tried `go test ./internal/runner`, but the broader local suite fails in existing Cypress/Jest tests because local JS test dependencies/paths are unavailable (`cypress` not found / Jest “No tests found”).

**Comment by pda** [comment]
@codex review

File: internal/runner/pytest.go [hunk 2]
```diff
@@ -120,7 +120,11 @@ func (p Pytest) Run(result *RunResult, testCases []plan.TestCase, retry bool) er
 	// Only rescue exit code 1 because it indicates a test failures.
 	// Ref: https://docs.pytest.org/en/7.1.x/reference/exit-codes.html
 	if exitError := new(exec.ExitError); errors.As(cmdErr, &exitError) && exitError.ExitCode() != 1 {
-		return cmdErr
+		// pytest exits 2 for collection errors, and buildkite-test-collector can
+		// still write tagged JSON results for those errors.
+		if p.useJUnit || exitError.ExitCode() != 2 {
+			return cmdErr
```
[the remark this answers is not in the document]
Comment (pda) [reply internal/runner/pytest.go:127]: 🤖 Fixed in b4fc68d. Exit-2 JSON is still parsed only far enough to detect tagged collection errors; if parsing does not produce `RunStatusError`, the run is marked with the original exit-2 command error so it remains terminal and cannot be retried into a false pass. Added `TestPytestRun_JSONExit2WithoutCollectionErrorIsTerminal` for this case.

Verified with focused runner and retry tests.

File: internal/runner/pytest.go [hunk 3]
```diff
@@ -120,7 +120,11 @@ func (p Pytest) Run(result *RunResult, testCases []plan.TestCase, retry bool) er
 	// Only rescue exit code 1 because it indicates a test failures.
 	// Ref: https://docs.pytest.org/en/7.1.x/reference/exit-codes.html
 	if exitError := new(exec.ExitError); errors.As(cmdErr, &exitError) && exitError.ExitCode() != 1 {
-		return cmdErr
+		// pytest exits 2 for collection errors, and buildkite-test-collector can
+		// still write tagged JSON results for those errors.
+		if p.useJUnit || exitError.ExitCode() != 2 {
```
[the remark this answers is not in the document]
Comment (pda) [reply internal/runner/pytest.go:126]: 🤖 Fixed in b4fc68d. Exit-2 JSON is still parsed so tagged collection errors can set `RunStatusError`, but an untagged exit-2 report now sets `result.error` to the original command error after parsing. That keeps interrupted/non-collection exit 2 terminal and prevents `runTestsWithRetry` from retrying or masking it.

Verified with:

```sh
go test ./internal/runner -run 'TestPytestRun_JSONExit2|TestPytestRunParseJSON_CollectionError|TestPytestRunParseJSON_EmptyScopeWithoutCollectionErrorTag|TestPytestPantsRun_JSONExit2'
go test ./internal/runner -run 'TestPytestRun_JSONExit2|TestPytestPantsRun_JSONExit2|TestPytestPantsRun_CommandFailed'
go test ./internal/command -run 'TestRunTestsWithRetry_RunResultError'
```

File: internal/runner/pytest_pants.go [hunk 4]
```diff
@@ -86,15 +86,7 @@ func (p PytestPants) Run(result *RunResult, testCases []plan.TestCase, retry boo
...
-		result.RecordTestResult(plan.TestCase{
-			Identifier: test.ID,
-			Format:     plan.TestCaseFormatExample,
-			Scope:      test.Scope,
-			Name:       test.Name,
-			// pytest can execute individual test using node id, which is a filename, classname (if any), and function, separated by `::`.
-			// Ref: https://docs.pytest.org/en/6.2.x/usage.html#nodeids
-			Path: fmt.Sprintf("%s::%s", test.Scope, test.Name),
-		}, test.Result)
+		recordPytestJSONTestResult(result, test)
```
[the remark this answers is not in the document]
Comment (pda) [reply internal/runner/pytest_pants.go:98]: 🤖 Fixed in b4fc68d. `pytest-pants` now mirrors the pytest JSON behavior for exit 2: it parses tagged collection-error JSON so the shared collection-error mapping is reached, while untagged exit-2 JSON is marked terminal with the original command error. Added pytest-pants coverage for both tagged collection-error exit 2 and untagged terminal exit 2.

Verified with focused runner tests.

**Comment by pda** [comment]
@codex review
